syft/format/common/cyclonedxhelpers/to_format_model.go

```diff
@@ -82,10 +83,12 @@ func toOSComponent(distro *linux.Release) []cyclonedx.Component {
 	if len(props) > 0 {
 		properties = &props
 	}
+	bomRef := fmt.Sprintf("os:%s@%s", distro.ID, distro.VersionID)
```

Comment:
I think we should check for cases when the ID or VersionID is missing -- I think I've seen this before, trying to collect some examples

Reply:
arch linux is an example of a missing version
```
docker run --rm -it --platform linux/amd64 archlinux:latest
[root@92ced9395309 /]# cat /etc/os-release
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
VERSION_ID=20240101.0.204074
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://bugs.archlinux.org/"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo
```

Reply:
If these are empty values, we would get something like: `os:@1.23`, `os:thing@` or `os:@`. These are all fairly clear to me that a name and/or version is missing and should be unique within the document, is there some other behavior that you'd expect?

Reply:
Small changes like `os:unknown` over `os:@1.23` or `os:@` and `os:thing` over `os:thing@`. I was originally thinking looking at fallback fields such as ID_LIKE, PRETTY_NAME, NAME, VERSION, etc... but I don't think that would be a good idea yet since some of these fields might give an incorrect indication of the identity (like ID_LIKE).

Reply:
I don't have a strong opinion on the verbiage; added some hopefully obviuous/clear things; test that show what it looks like are: https://github.com/anchore/syft/pull/2634/files#diff-ef679fb96ee08c5fcd0fe9323424af7502f51d170aa8571f861632027749ef0cR292-R315